source/binary.cpp

```diff
@@ -752,7 +753,13 @@ spv_result_t spvBinaryParse(const spv_const_context context, void* user_data,
                             spv_parsed_header_fn_t parsed_header,
                             spv_parsed_instruction_fn_t parsed_instruction,
                             spv_diagnostic* diagnostic) {
-  Parser parser(context, user_data, parsed_header, parsed_instruction);
+  if (!context->consumer && !diagnostic) return SPV_ERROR_INVALID_DIAGNOSTIC;
```

Comment:
I don't like this.
I should be able to use the binary parser without collecting detailed feedback. I could trust the binaries I'm passing in.